### ISSUE TITLE
FR: different alert recipients

### ISSUE THREAD

**Issue by tkoeberl** [guilhemmarchand/trackme#108]
Would be cool to be able to have different recipients for different index:sourcetype combinations.
Maybe like the "Data source week days monitoring:" part.

**Comment by guilhemmarchand** [comment]
Hi @tkoeberl 

Thanks for taking the time to share this, I had the question somehow earlier and will consider it.

Guilhem

**Comment by guilhemmarchand** [comment]
@tkoeberl 

In my opinion, here is what you would do to achieve your goal:

For ref: https://docs.splunk.com/Documentation/Splunk/8.0.5/Alert/EmailNotificationTokens

Basically, what you would is:

- Create a CSV file base lookup where you maintain the ownership of the team responsible for a given data source
- Create a CSV file base lookup which mapps the teams to the email address
- Customise the OOTB alert to get enrichment from these 2 lookups, and use the value of the recipient via a token

**Consider the following example:**

data_source:

`
network:firewall
network:loadbalancer
osnix:syslog
oswin:WinEventLogs
`

Create a first lookup which does

`
data_source,owner
network:firewall,team1
network:loadbalancer,team1
osnix:syslog,team2
oswin:WinEventLogs,team2
`

Create a second lookup which does

`
team,email
team1,team1@mydomain.com
team2,team2@mydomain.com
`

Then you will customise the alert to add at the very end:

`
| lookup team_owner.csv data_name OUTPUT owner as team
| lookup team_inventory.csv team OUTPUT email
`

Finally, in your email alert action, in the recipient box you would use:

$result.email$

Instead of the hard coded email address.
The reason why you would create 2 lookups will be to avoid repeating the email addresses again and again, and be smarter with a dual mapping.

Guilhem

**Comment by tkoeberl** [comment]
@guilhemmarchand :
that would work, no question. 
But you already have a fancy configuration UI ;) If you do not implement it to have it native in your app, I would implement it like that. We already have lookups, that can be used for the job.
thanks,
-thomas

**Comment by guilhemmarchand** [comment]
Thanks @tkoeberl  ha ha that's true the UI is fancy ;-) I will keep that in the list to see if a more native approach from the app point of view is possible

**Comment by guilhemmarchand** [comment]
@tkoeberl 

After reviewing this enhancement request, I think this would be best to let this handled by any usual Splunk way as exposed earlier, and focuss the app on the next upcoming very hot features and value ;-)

Please feel free if you have any comment, and I will be more than glad to have anymore suggestions from you.

Guilhem

**Comment by tkoeberl** [comment]
@guilhemmarchand 
No prob. already did that ;) Only wanted to make sure if it is woth to do it.

**Comment by guilhemmarchand** [comment]
Sure thanks

Do you have a testing environment @tkoeberl ?
I am looking for beta testers for the new release and specially for a new hot feature in Trackme:

The new Beta version is the version 1.2.21, in Git: (testing branch)

https://github.com/guilhemmarchand/trackme/tree/testing

The new very hot feature I am contacting you for is in relation with what I called the data sampling and event format recognition engine, in short it is an automated workflow that processes samples of the raw events from the data sources, perform event format recognition over time to detect changes in the format or misbehaviour such as multiple format in the same sourcetype.

The documentation of the testing branch is here:

https://trackme.readthedocs.io/en/testing/

Specially regarding the new feature:

https://trackme.readthedocs.io/en/testing/userguide.html#data-sampling-and-event-formats-recognition

I am especially interested in validating the performance and the behaviour of the search at scale, I have introduced a concept in the search that should in theory limit any performance issues and avoid long run time of the search, dynamically based on the number of entities to be processed, the historical performance of the tracker to determine a rate per sec of entity processing and apply a simple formula.

So in theory, the first time the tracker runs, it samples a minor number of data sources, recycles the performance stats, to evaluate the number of entities to processes to respect a maximal run time in seconds one would want to allow the search to (defaults to 120 seconds)
Basically this would allow the feature to handle the feature in a ideal way according to the real World performance of the search.

If you have the chance to test and share your feedback this would be lovely ;-)

Thanks

Guilhem

**Comment by tkoeberl** [comment]
@guilhemmarchand :
not sure if I can test it, but for me it sound it needs to be tested on real data. > means prod env.  
What could it break?

**Comment by guilhemmarchand** [comment]
No problem @tkoeberl 
It's more about detecting issues, getting feedback on the feature and proof testing.
But that is ok, I have some beta testing it currently, thanks
